src/scripts/main.js

```diff
@@ -1,3 +1,315 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+const table = document.querySelector('table');
+const headers = table.querySelectorAll('th');
+const tBodies = table.querySelector('tbody');
+
+let selectedRow = null;
+let activeCell = null;
+
+const directions = Array.from(headers).map(function(header) {
+  return '';
+});
+
+headers.forEach(header => {
+  header.addEventListener('click', () => {
+    const index = header.cellIndex;
+    const rows = table.tBodies[0].querySelectorAll('tr');
+    const rowsArray = [...rows];
+
+    const direction = directions[index] || 'asc';
+
+    const multiplier = (direction === 'asc') ? 1 : -1;
+
+    rowsArray.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[index].textContent;
+      const cellB = b.querySelectorAll('td')[index].textContent;
+
+      function num(str) {
+        const hasDigits = /\d/.test(str);
+
+        return hasDigits ? +str.replace(/\D/g, '') : str;
+      };
+
+      switch (true) {
```

Comment:
you shouldn't use switch is in switch block you pass true, here should be some condition